.github/workflows/publish-api-doc.yml

```diff
@@ -0,0 +1,63 @@
+
+# This generates Doxygen autodocs for the C++ API (for the libyui base lib)
+# and pushes them to the "gh-pages" branch which publishes them
+# to the project's GitHub pages at https://libyui.github.io/libyui/api-doc/
+#
+# See also
+#
+#   https://gist.github.com/shundhammer/ed359db0d9329d4db551528256060d2a
+#
+# GitHub actions syntax:
+#   https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+name: publish-api-doc
+
+# Trigger this when a pull request is merged (which implies pushing to master).
+on:
+  push:
+    branches:
+      # master
+      - main
+
+      # push to this branch to trigger a test run of this action
+      # (which does the real publishing, but you don't need to merge a PR)
+      - api-doc-test
+
+jobs:
+  api-doc:
+    runs-on: ubuntu-latest
+    container: registry.opensuse.org/devel/libraries/libyui/containers/libyui-devel:latest
+
+    steps:
+
+    - name: Git Checkout
+      uses: actions/checkout@v2
+
+    - name: Create clean gh-pages branch
+      run:  git checkout -b gh-pages
+
+    - name: Generate autodocs for libyui base lib
+      run:  make -C libyui -f Makefile.repo doc
+
+    - name: Move generated autodocs to target directory
+      run:  mv libyui/build/doc/html api-doc
+
+    - name: Add generated autodocs to Git repo in the gh-pages branch
+      run:  >
```

Comment:
I usually use the `|` multiline variant, it keeps the newlines. Then you do not need the trailing `;` on each line but the last one, that's more error proof. See https://yaml-multiline.info/

Reply:
I tried the '|' version first and I found that I did need _both_ a trailing backslash and the semicolon. I'll check again.

Reply:
OK; I tried again, and it works with `|`. Changed.